### PR TITLE
Added custom Claude command + XNACK command support

### PR DESCRIPTION
**Pull request by vladvildanov**
### Description of change

This PR contains:

1. New custom Claude command that enrich agent with necessary context and provides a clear guideline on how to add a support for given Redis command. Also it contains a command specification template to be used to provide custom command specification.
2. Support for XNACK command fully added using new Claude command.
3. Update to `CONTRIBUTING.md` about AI agent contributions.

Execute new command via CLI interface:

**Augment CLI**

```
auggie command add-new-command $path_to_specification
```

**Claude CLI**

```
claude /add-new-command $path_to_specification
```

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Do tests and lints pass with this change?
- [ ] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [ ] Is the new or changed code fully tested?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Is there an example added to the examples folder (if applicable)?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

---

> [!NOTE]
> **Medium Risk**
> Introduces a new public client API method that constructs a low-level Redis command; incorrect argument encoding/order could cause runtime errors, though coverage is mitigated by added integration tests.
> 
> **Overview**
> Adds first-class client support for Redis Streams `XNACK`, including sync/async overloads, argument validation (mode, required IDs, non-negative `retrycount`), and support for `RETRYCOUNT`/`FORCE` options.
> 
> Expands integration test coverage (sync + asyncio, Redis >= 8.7.2) for `XNACK` modes and edge cases, and adds contributor tooling/docs: a Claude “add-new-command” skill plus a command specification template and a Redis command implementation guide; `CONTRIBUTING.md` and the spelling wordlist are updated accordingly.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3ffd26bf677f0863f112b731b3bbc5ce33d9c89f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: redis/commands/core.py [hunk 1]
```diff
@@ -6801,6 +6801,78 @@ def xlen(self, name: KeyT) -> int | Awaitable[int]:
...
+        pieces: list = [name, groupname, mode, "IDS", len(ids)]
+        pieces.extend(ids)
+
+        if retrycount is not None:
+            if retrycount < 0:
+                raise DataError("XNACK retrycount must be >= 0")
+            pieces.extend([b"RETRYCOUNT", retrycount])
+
+        if force:
+            pieces.append(b"FORCE")
```
[the remark this answers is not in the document]
Comment (vladvildanov) [reply redis/commands/core.py:6872]: Types are defined in method arguments, there's no point to do additional type check

File: specs/redis_commands_guide.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+### Binary representation
+
+Our SDK can be configured with `decode_response=False` option which means that command response will be returned
+as bytes, as well as we allow to pass bytes arguments instead of strings. For this purpose, we define a custom types
+like `KeyT`, `EncodableT`, etc. in `redis/typing.py`.
+
+## Protocols compatibility
+
+SDK supports two types of Redis wire protocol: RESP2 and RESP3. And aims to provide a compatibility between them
+for seamless user experience. However, there are some differences in types that defined by these protocols.
```
Comment (petyaslavova) [inline specs/redis_commands_guide.md:46]: types that defined --> types that are defined

File: specs/redis_commands_guide.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+Because, RESP3 introduce new types that wasn't previously supported by RESP2, we're aiming for forward compatibility
```
Comment (petyaslavova) [inline specs/redis_commands_guide.md:48]: types that wasn't  --> types that weren't

File: specs/redis_commands_guide.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+and ensure that all new types supported by RESP3 can be used with RESP2. In most cases the semantic of RESP3 can be
+easily recognized and converting existing RESP2 response in RESP3 is a matter of parsing strategy.
+
+All new commands should be added with RESP3 in mind and by default should hide protocol incompatibility from user.
+
+To understand how does Redis types defined by RESP2 and RESP3 protocol maps to Python types,
+see `redis/_parsers/resp2.py` and `redis/_parsers/resp3.py`.
+
+Parsers are responsible for RESP protocol parsing. However, protocols compatibility is achieved by 2nd layer
+parsing as `response_callbacks` defined in `redis/_parsers/helpers.py` and can be extended/updated on client level.
```
Comment (petyaslavova) [inline specs/redis_commands_guide.md:58]: For modules the callbacks are defined usually in module's `utils.py` file

File: tests/test_asyncio/test_commands.py [hunk 5]
```diff
@@ -4301,6 +4301,114 @@ async def test_xlen(self, r: redis.Redis):
         await r.xadd(stream, {"foo": "bar"})
         assert await r.xlen(stream) == 2
 
+    @skip_if_server_version_lt("8.8.0")
+    async def test_xnack_silent(self, r: redis.Redis):
```
Comment (petyaslavova) [inline tests/test_asyncio/test_commands.py:4305]: The r type hint should also allow RedisCluster

Comment (petyaslavova) [reply tests/test_asyncio/test_commands.py:4305]: The RC versions are usually coming with a little different version - for 8.8 we have 8.7.something. If we add these tests with this restriction, they won't actually get executed until we have a GA

File: specs/redis_commands_guide.md (hunk 2 above)
Comment (vladvildanov) [reply specs/redis_commands_guide.md]: Fixed

File: specs/redis_commands_guide.md (hunk 3 above)
Comment (vladvildanov) [reply specs/redis_commands_guide.md]: Fixed

File: specs/redis_commands_guide.md (hunk 4 above)
Comment (vladvildanov) [reply specs/redis_commands_guide.md]: Added

File: tests/test_asyncio/test_commands.py (hunk 5 above)
Comment (vladvildanov) [reply tests/test_asyncio/test_commands.py]: Updated both

**Review by petyaslavova — APPROVED** [review]
LGTM.
